runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/Work.java

```diff
@@ -125,6 +125,11 @@ boolean isStuckCommittingAt(Instant stuckCommitDeadline) {
         && currentState.startTime().isBefore(stuckCommitDeadline);
   }
 
+  boolean isRetryOf(WorkDedupeKey workDedupeKey) {
```

Comment:
maybe instead put this method on WorkDedupeKey and have a method vending WorkDedupKey from Work?

Reply:
done